library/control/src/modules/ProductControl.rb

```diff
@@ -1336,16 +1337,47 @@ def RunFrom(from, allow_back)
           end
         end
 
-        Hooks.run("before_#{step_name}")
+        client_name = getClientName(step_name, step_execute)
 
-        result = WFM.CallFunction(getClientName(step_name, step_execute), args)
+        # Check if client exist before continuing
+        if WFM.ClientExists(client_name)
+          Hooks.run("before_#{step_name}")
 
-        # this code will be triggered before the red pop window appears on the user's screen
-        Hooks.run("installation_failure") if result == false
+          result = WFM.CallFunction(client_name, args)
 
-        result = Convert.to_symbol(result)
+          # This code will be triggered before the red pop window appears on the user's screen
+          Hooks.run("installation_failure") if result == false
 
-        Hooks.run("after_#{step_name}")
+          result = result.to_sym
+
+          Hooks.run("after_#{step_name}")
+        else
+          # Client not found. Ask the user if want to continue (related to bsc#1180954)
+          Builtins.y2error("Client '%1' not found", client_name)
+
+          text = format(
+            # TRANSLATORS: Message warning the user that a client is missing where %{client} is
+            # replaced by the client name (e.g. "registration", "user")
+            _(
+              "Something went wrong and the expected '%{client}' dialog was not found.\n\n" \
+              "Would you like to continue anyway?"
+            ),
+            client: client_name
+          )
+
+          continue = Yast2::Popup.show(text, buttons: :yes_no) == :yes
```

Comment:
Maybe we should something like  

```ruby
Yast2::Popup.show(text, buttons: { yes: Label.ContinueButton, no:  Label.AbortButton) == :yes
```

here to display the _Abort_ and _Continue_ buttons instead.

Thoughts?

Reply:
yes, continue/abort looks better for me. Question is if we should not use Report here as code is used also in other places and not just in firstboot.

Reply:
So, maybe to apply this control/check when running in Stage.firstboot?

Reply:
this check is useful also for other modes, so I would not do it. Popup is OK as in other cases it usually means bug in our control.xml which should be well tested by openqa.